vochain/indexer/archive.go

```diff
@@ -61,17 +61,33 @@ func (idx *Indexer) ImportArchive(archive []*ArchiveProcess) ([]*ArchiveProcess,
 		} else {
 			continue
 		}
-		creationTime := time.Now()
-		if p.StartDate != nil {
-			creationTime = *p.StartDate
+
+		// For backward compatibility, we try to fetch the start/end date from multiple sources.
+		// If not found, we calculate them from the block count and the default block time.
+		startDate := p.ProcessInfo.StartDate
+		if startDate.IsZero() {
+			if p.StartDate != nil {
+				startDate = *p.StartDate
+			} else {
+				// Calculate startDate equal to time.Now() minus defaultBlockTime*p.ProcessInfo.BlockCount
+				startDate = time.Now().Add(-defaultBlockTime * time.Duration(p.ProcessInfo.BlockCount))
+			}
+		}
+		endDate := p.ProcessInfo.EndDate
+		if endDate.IsZero() {
+			// Calculate endDate equal to startDate plus defaultBlockTime*p.ProcessInfo.BlockCount
+			endDate = startDate.Add(defaultBlockTime * time.Duration(p.ProcessInfo.BlockCount))
 		}
+
 		// Create and store process in the indexer database
 		procParams := indexerdb.CreateProcessParams{
```

Comment:
why are we not storing the ManuallyEnded bool anywhere?

Reply:
ah, now i see, ManuallyEnded is calculated based on `pi.EndBlock < pi.StartBlock+pi.BlockCount` , sorry for the noise